server/src/main/java/io/pravega/schemaregistry/server/rest/RestServer.java

```diff
@@ -0,0 +1,103 @@
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.server.rest;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import io.pravega.common.LoggerHelpers;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.core.UriBuilder;
+
+import io.pravega.common.auth.JKSHelper;
+import io.pravega.schemaregistry.server.rest.resources.GroupResourceImpl;
+import io.pravega.schemaregistry.server.rest.resources.PingImpl;
+import io.pravega.schemaregistry.server.rest.resources.SchemaResourceImpl;
+import io.pravega.schemaregistry.service.SchemaRegistryService;
+import lombok.extern.slf4j.Slf4j;
+import org.glassfish.grizzly.GrizzlyFuture;
+import org.glassfish.grizzly.http.server.HttpServer;
+import org.glassfish.grizzly.ssl.SSLContextConfigurator;
+import org.glassfish.grizzly.ssl.SSLEngineConfigurator;
+import org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.ServerProperties;
+
+/**
+ * Netty REST server implementation.
+ */
+@Slf4j
+public class RestServer extends AbstractIdleService {
+    private final String objectId;
+    private final ServiceConfig restServerConfig;
+    private final URI baseUri;
+    private final ResourceConfig resourceConfig;
+    private HttpServer httpServer;
+
+    public RestServer(SchemaRegistryService registryService, ServiceConfig restServerConfig) {
+        this.objectId = "RestServer";
+        this.restServerConfig = restServerConfig;
+        this.baseUri = UriBuilder.fromUri("http://" + restServerConfig.getHost()).port(restServerConfig.getPort()).build();
+
+        final Set<Object> resourceObjs = new HashSet<>();
+        resourceObjs.add(new PingImpl());
+        resourceObjs.add(new GroupResourceImpl(registryService, restServerConfig, executor()));
+        resourceObjs.add(new SchemaResourceImpl(registryService, restServerConfig, executor()));
+
+        final RegistryApplication application = new RegistryApplication(resourceObjs);
+        this.resourceConfig = ResourceConfig.forApplication(application);
+        this.resourceConfig.property(ServerProperties.BV_SEND_ERROR_IN_RESPONSE, true);
+
+        // Register the custom JSON parser.
+        this.resourceConfig.register(new CustomObjectMapperProvider());
+    }
+
+    /**
+     * Start REST service.
+     */
+    @Override
+    protected void startUp() {
+        long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
+        try {
+            log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator contextConfigurator = new SSLContextConfigurator();
+                contextConfigurator.setKeyStoreFile(restServerConfig.getTlsKeyFilePath());
```

Comment:
This comment applies elsewhere, but thought should mention it here so that I don't miss. 

Please modify the `getTlsKeyFilePath()` method to something like `getServerKeyStoreFilePath()` as it really represents the keystore containing the server's private key as well as the public-key certificate. The current method appears to indicate that it contains the private key alone, which is misleading. 

I know we have used the same name in Pravega codebase, we should change it there too sometime.